### DISCUSSION TITLE
Error: no provider exists with the given name

### DISCUSSION THREAD

**Discussion by akbarahs** [terra-farm/terraform-provider-virtualbox#136] [Support]
This is my terraform main script

    terraform {
      required_providers {
        virtualbox = {
          source = "terra-farm/virtualbox"
          version = "0.2.2-alpha.1"
        }
      }
    }

    provider "virtualbox" {
      # Configuration options
    }

    resource "virtualbox_vm" "node" {
        count = 2
        name = "${format("node-%02d", count.index+1)}"
         url = "https://vagrantcloud.com/ubuntu/boxes/xenial64/versions/20180420.0.0/providers/virtualbox.box"
        image = "./virtualbox.box"
        cpus = 2
        memory = "512 mib"
       // user_data = "${file("user_data")}"

        network_adapter {
        type = "bridged"
        host_interface="en0"
        }
    }

and this where the plugins are available

    PS C:\Users\Heisenberg\.terraform.d\plugins> dir
    Directory: C:\Users\Heisenberg\.terraform.d\plugins
    Mode                 LastWriteTime         Length Name
    ----                 -------------         ------ ----
    d-----        13-02-2022  12:05 PM                terraform-provider-virtualbox
    d-----        13-02-2022  12:05 PM                terraform-provider-virtualbox_v0.2.1
    d-----        13-02-2022  12:05 PM                windows_amd64
    PS C:\Users\Heisenberg\.terraform.d\plugins>

and also here

    PS C:\Users\Heisenberg\AppData\Local\.terraform.d\plugins> dir
    Directory: C:\Users\Heisenberg\AppData\Local\.terraform.d\plugins
    Mode                 LastWriteTime         Length Name
    ----                 -------------         ------ ----
    d-----        13-02-2022  12:05 PM                terraform-provider-virtualbox
    d-----        13-02-2022  12:05 PM                terraform-provider-virtualbox_v0.2.2-alpha.1
    d-----        13-02-2022  12:17 PM                windows_amd64
    PS C:\Users\Heisenberg\AppData\Local\.terraform.d\plugins>

but still terraform couldnt find the plugins. I followed your beautiful answer but I am still unable to make it work. Please help a brother out.

    PS E:\Practice-Terraform\VirtualBox VM> terraform init
    
    Initializing provider plugins...
    - Checking for available provider plugins...
    
    Provider "virtualbox" not available for installation.
    
    A provider named "virtualbox" could not be found in the Terraform Registry.
    
    This may result from mistyping the provider name, or the given provider may
    be a third-party provider that cannot be installed automatically.
    
    In the latter case, the plugin must be installed manually by locating and
    downloading a suitable distribution package and placing the plugin's executable
    file in the following directory:
        terraform.d/plugins/windows_amd64
    
    Terraform detects necessary plugins by inspecting the configuration and state.
    To view the provider versions requested by each module, run
    "terraform providers".
    
    
    Warning: Skipping backend initialization pending configuration upgrade
    
    The root module configuration contains errors that may be fixed by running the 
    configuration upgrade tool, so Terraform is skipping backend initialization. See 
    below for more information.
    
    
    Error: no provider exists with the given name

Thank you

**Comment by VoyTechnology** [comment]
What is the version of Terraform that you are using? Does the error occur with the latest version of Terraform as well?
